### DISCUSSION TITLE
Modify the @info() decorator to format the Resolve Info for prisma selection set

### DISCUSSION THREAD

**Discussion by YassinEldeeb** [MichalLytek/type-graphql#948] [Q&A] [answered]
I'm using paljs specifically their `PrismaSelect` plugin for converting graphql resolve info to prisma selection set then passing it to prisma, but I keep repeating this single line in each resolver `const { select } = new PrismaSelect(info).value` and it's really annoying.

![Screenshot 2021-07-12 004412](https://user-images.githubusercontent.com/72823042/125211939-539c4500-e2aa-11eb-8be1-e5fb208fa054.png)

I thought of making a global middleware which overwrite the info to be the prisma selection set, but the problem was that the global midddleware run not only for the resolvers but for the field resolvers too which is not what I wanted at all and even with `@UseMiddlware()` attached to a specific resolver the info value isn't overwritten.

so can I extend the `@Info()` functionality to run the `PrismaSelect` utility and then return the info to me in the resolvers or making a custom decorator like `@Select` which I can use it in resolvers as a parameter, and this decorator takes the info and format it for prisma selection set.

or there's another way for achieving what I want rather than what I mentioned.

**Comment by MichalLytek** [comment, marked as answer]
https://typegraphql.com/docs/custom-decorators.html#parameter-decorators

**Reply by YassinEldeeb** [reply]
That's literally what I wanted, Thanks!
